kv/kvserver: TestTenantRateLimiter failed

kv/kvserver.TestTenantRateLimiter [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9281248?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9281248?buildTab=artifacts#/) on master @ [3f113fc1a34e5bb8922298fbbb17c0c7f7ede4ac](https://github.com/cockroachdb/cockroach/commits/3f113fc1a34e5bb8922298fbbb17c0c7f7ede4ac):

Fatal error:

```
panic: test timed out after 59m55s

```
Stack: 

```
goroutine 109556 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2036 +0x88
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x38
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestTenantRateLimiter
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/645d36459c8d7b7939924714a555a08a/logTestTenantRateLimiter3873207494
    test_log_scope.go:79: use -show-logs to present logs inline
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantRateLimiter.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-26051

kv/kvserver.TestTenantRateLimiter [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9281996?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9281996?buildTab=artifacts#/) on master @ [c1b1b4ce6dd871e0b26b18d0801094e3191ec1e9](https://github.com/cockroachdb/cockroach/commits/c1b1b4ce6dd871e0b26b18d0801094e3191ec1e9):

Fatal error:

```
panic: test timed out after 59m55s

```
Stack: 

```
goroutine 135624 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2036 +0x88
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x38
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestTenantRateLimiter
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/645d36459c8d7b7939924714a555a08a/logTestTenantRateLimiter2056985728
    test_log_scope.go:79: use -show-logs to present logs inline
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantRateLimiter.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvserver.TestTenantRateLimiter [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9283136?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9283136?buildTab=artifacts#/) on master @ [143b63a6a27aeb286637dd2e5abddafdf0c51874](https://github.com/cockroachdb/cockroach/commits/143b63a6a27aeb286637dd2e5abddafdf0c51874):

Fatal error:

```
panic: test timed out after 59m55s

```
Stack: 

```
goroutine 139884 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2036 +0x88
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x38
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestTenantRateLimiter
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/645d36459c8d7b7939924714a555a08a/logTestTenantRateLimiter111601478
    test_log_scope.go:79: use -show-logs to present logs inline
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantRateLimiter.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/kv/kvserver/BUILD.bazel
+++ b/pkg/kv/kvserver/BUILD.bazel
@@ -345,6 +345,7 @@ go_test(
         "//pkg/config",
         "//pkg/config/zonepb",
         "//pkg/gossip",
+        "//pkg/jobs",
         "//pkg/keys",
         "//pkg/keyvisualizer",
         "//pkg/kv",

--- a/pkg/kv/kvserver/client_tenant_test.go
+++ b/pkg/kv/kvserver/client_tenant_test.go
@@ -25,6 +25,7 @@ import (
 
 	"github.com/cockroachdb/cockroach/pkg/base"
 	_ "github.com/cockroachdb/cockroach/pkg/ccl" // for tenant functionality
+	"github.com/cockroachdb/cockroach/pkg/jobs"
 	"github.com/cockroachdb/cockroach/pkg/keys"
 	"github.com/cockroachdb/cockroach/pkg/keyvisualizer"
 	"github.com/cockroachdb/cockroach/pkg/kv/kvpb"
@@ -172,7 +173,11 @@ func TestTenantRateLimiter(t *testing.T) {
 				},
 			},
 			KeyVisualizer: &keyvisualizer.TestingKnobs{SkipJobBootstrap: true},
+			JobsTestingKnobs: &jobs.TestingKnobs{
+				DisableAdoptions: true,
+			},
 			UpgradeManager: &upgradebase.TestingKnobs{
+				DontUseJobs:                       true,
 				SkipJobMetricsPollingJobBootstrap: true,
 				SkipAutoConfigRunnerJobBootstrap:  true,
 			},
@@ -183,6 +188,13 @@ func TestTenantRateLimiter(t *testing.T) {
 	ts, err := s.StartTenant(ctx, base.TestTenantArgs{
 		TenantID: tenantID,
 		TestingKnobs: base.TestingKnobs{
+			JobsTestingKnobs: &jobs.TestingKnobs{
+				DisableAdoptions: true,
+			},
+			// DisableAdoptions needs this.
+			UpgradeManager: &upgradebase.TestingKnobs{
+				DontUseJobs: true,
+			},
 			SpanConfig: &spanconfig.TestingKnobs{
 				// Disable the span reconciler because it performs tenant KV requests
 				// that interfere with our operation counts below.
@@ -224,6 +236,11 @@ func TestTenantRateLimiter(t *testing.T) {
 	// tooManyWrites is a number of writes which definitely exceed the burst
 	// limit.
 	tooManyWrites := int(cfg.Burst/writeCostLower) + 2
+
+	// This test shouldn't take forever. If we're going to fail, better to
+	// do it in minutes than in an hour.
+	ctx, cancel := context.WithTimeout(ctx, 2*time.Minute)
+	defer cancel()
 
 	// Make sure that writes to the system tenant don't block, even if we
 	// definitely exceed the burst rate.

